.circleci/config.yml

```diff
@@ -0,0 +1,426 @@
+version: 2.1
+
+defaults: &defaults
+  working_directory: ~/repo
+  machine:
+    image: circleci/classic:latest
+  environment:
+    TERM: dumb
+
+commands:
+
+  generate-cache-key:
+    description: "Generate cache key from project files"
+    steps:
+      # Cache key should consider not only build.sbt but all
+      # project configurations. For example, if we update a
+      # dependency, or a sbt plugin, or sbt version, the cache
+      # needs to be updated.
+      # Circle does not have out-of-the-box a way to checksum
+      # a directory, so this command creates a file that can
+      # later be used to generate the cache key.
+      - run:
+          name: "Generate cache key"
+          command: |
+            find ./project -type f -exec md5sum {} \; > cache-key.txt
+            md5sum build.sbt >> cache-key.txt
```

Comment:
IIUC this line should be copy/pasted for any file in `project/`? Any reason you are not doing it here already?